source/particles/particle_handler.cc

```diff
@@ -1558,9 +1567,15 @@ namespace Particles
             std::set<unsigned int> cell_to_neighbor_subdomain;
             for (const unsigned int v : cell->vertex_indices())
               {
-                cell_to_neighbor_subdomain.insert(
-                  vertex_to_neighbor_subdomain[cell->vertex_index(v)].begin(),
-                  vertex_to_neighbor_subdomain[cell->vertex_index(v)].end());
+                if (vertices_with_ghost_neighbors.find(cell->vertex_index(v)) !=
+                    vertices_with_ghost_neighbors.end())
+                  {
+                    cell_to_neighbor_subdomain.insert(
+                      vertices_with_ghost_neighbors.at(cell->vertex_index(v))
+                        .begin(),
+                      vertices_with_ghost_neighbors.at(cell->vertex_index(v))
+                        .end());
+                  }
```

Comment:
Actually I just realized we can cut the cost of this operation in half the following way (by storing the result of .find and reusing it if there are ghost domains):

```suggestion
                const auto vertex_ghost_neighbors = vertices_with_ghost_neighbors.find(cell->vertex_index(v));
                if (vertex_ghost_neighbors !=
                    vertices_with_ghost_neighbors.end())
                  {
                    cell_to_neighbor_subdomain.insert(
                      vertex_ghost_neighbors->second.begin(),
                      vertex_ghost_neighbors->second.end());
                  }
```